src/builder.rs

```diff
@@ -91,7 +91,10 @@ pub enum BuilderError {
 ///
 /// This trait is already implemented generically for the [`DeviceBuilder`]
 /// and implementing it should be avoided since it has no practical use.
-pub trait DeviceSdkBuild<S, C> {
+pub trait DeviceSdkBuild<S, C>
```

Comment:
Is it possible to  move `S and C` as Associated types to improve the overall readability ?

```suggestion
pub trait DeviceSdkBuild {
    type S;
    type C;
```

Reply:
I'm unsure if this approach will work, as it would require the builder to implement this trait for a specific associated type, rather than being generic. The implementation could be for some generic parameter, but at that point, the code would essentially be the same as having generics in the trait, since we cannot remove them from the `DeviceBuilder` struct.

Here's an example:

<details title="example">

```rust

pub trait DeviceSdkBuild {
  type Store;
  type Conn;
}

impl<S,C> DeviceSdkBuilder for DeviceBuilder<S,C> {
  type Store = S;
  type Conn = C;
// ...
}

```

</details>

Also, we use generics to avoid exposing the underlying private types. However, implementing this trait would still expose those types.